server/src/main/java/org/elasticsearch/common/util/BigDoubleArray.java

```diff
@@ -24,6 +26,12 @@
  */
 final class BigDoubleArray extends AbstractBigArray implements DoubleArray {
 
+    static {
```

Comment:
Do we actually need this here as well? Maybe we can just make a static method for this somewhere at least since we really only use it once? Or put this in a bootstrap check? Seems strange to duplicate this check doesn't it?

Reply:
> Or put this in a bootstrap check?

I like this idea. I will attempt this in a followup pr.

Reply:
I've opened this pr for this change: #91801